src/UartSerial.cpp

```diff
@@ -104,22 +102,30 @@ void UartSerial::begin(uint32_t speed_, size_t config_)
 
 void UartSerial::end()
 {
-  if(_fd != -1){
-    flush();
-    // Restore the original settings
-    tcsetattr(_fd, TCSANOW, &_tio_original_config);
-    close(_fd);
-    _fd = -1;
-  }
+  if(_fd == -1) return;
+  
+  flush();
```

Comment:
Added return code checking to flush() but I think we should continue to call it this way. That way the implementation for flush can change without affecting the code in end().